packages/mapviewer/src/modules/infobox/components/FeatureDetail.vue

```diff
@@ -22,38 +22,63 @@ const { feature } = defineProps({
 const { t } = useI18n()
 
 const store = useStore()
-const hasFeatureStringData = computed(() => typeof feature?.data === 'string')
-const popupDataCanBeTrusted = computed(() => feature.popupDataCanBeTrusted)
+const hasFeatureStringData = computed(() => typeof props.feature?.data === 'string')
+const popupDataCanBeTrusted = computed(() => props.feature?.popupDataCanBeTrusted)
 
 const coordinateFormat = computed(() => {
     return allFormats.find((format) => format.id === store.state.position.displayedFormatId) ?? null
 })
 const sanitizedFeatureDataEntries = computed(() => {
-    if (hasFeatureStringData.value || !feature?.data) {
+    if (hasFeatureStringData.value || !props.feature?.data) {
         return []
     }
-    return Object.entries(feature.data)
-        .filter(([_, value]) => value) // filtering out null values
+    return Object.entries(props.feature?.data)
+        .filter(([_, value]) => value) // Filtering out null values
         .map(([key, value]) => [
             key,
             sanitizeHtml(value, key === 'description'),
-            getIframeHosts(value),
+            key === 'description' ? getIframeHosts(value) : [],
         ])
 })
 function sanitizeHtml(htmlText, withIframe = false) {
+    const blockedExternalContentString = t('blocked_external_content')
+
+    // Replacing possibly malicious code with `blockedExternalContentString` instead of removing it
+    function handleNode(node, attribute) {
+        try {
+            const url = new URL(node.getAttribute(attribute))
+            const ext = url.pathname.split('.').pop().toLowerCase()
+            if (BLOCKED_EXTENSIONS.includes(ext)) {
+                node.outerHTML = blockedExternalContentString
+                return true
+            }
+        } catch (error) {
+            log.error(error)
+            node.outerHTML = blockedExternalContentString
+            return true
+        }
+        return false
+    }
+
     DOMPurify.addHook('afterSanitizeAttributes', function (node) {
-        // set all elements owning target to target=_blank
-        if ('target' in node) {
+        // Check the A tag and add target="_blank" and rel="noopener noreferrer" to prevent tabnabbing
+        if (node.tagName === 'A') {
             node.setAttribute('target', '_blank')
-            node.setAttribute('rel', 'noopener')
+            node.setAttribute('rel', 'noopener noreferrer')
+            if (handleNode(node, 'href')){
+                return
```

Comment:
is it really necessary to return something from `handleNode`?
It doesn't really matter if `true` or `false` is returned, it will eventually go to the end of this function anyway.

IMHO `handleNode` could return nothing and that would also work as intended.

Reply:
agree.